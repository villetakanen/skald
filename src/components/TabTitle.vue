<template>
  <div class="tabtitle">
    <h1
      v-if="sub"
      :class="`subtopic ${titleColorClass}`">{{sub}}</h1>
    <h1
      v-if="topic"
      :class="`topic ${titleColorClass}`">{{topic}}</h1>
  </div>
</template>

<script>
export default {
  props: [
    'topic',
    'sub'
  ],
  computed: {
    titleColorClass () {
      return this.$store.getters['site/titleColorClass']()
    }
  }
}
</script>

<style lang="scss" scoped>
div.tabtitle{
  margin: 0;
  padding: 0;
  h1 {
    font-weight: 300;
    margin: 0;
    padding: 0;
  }
  h1+h1 {margin-top: 4px}
  h1.subtopic{
    font-size: 16px;
    line-height: 16px;
  }
  h1.topic{
    font-size: 32px;
    line-height: 32px;
  }
}
</style>
