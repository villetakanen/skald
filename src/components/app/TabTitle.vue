<template>
  <div class="tabtitle">
    <h1
      v-if="sub"
      :class="`subtopic ${titleColorClass}`">
      <template v-if="!sublink">{{ sub }}</template>
      <router-link
        v-if="sublink"
        :to="sublink"
        :class="titleColorClass"
        >{{ sub }}</router-link></h1>
    <h1
      v-if="topic"
      :class="`topic ${titleColorClass}`">
      <template v-if="!link">{{ topic }}</template>
      <router-link
        v-if="link"
        :to="link"
        :class="titleColorClass"
        >{{ topic }}</router-link></h1>
  </div>
</template>

<script lang="ts">
import VueCompositionApi, { defineComponent, computed } from '@vue/composition-api'
import { useSite } from '@/lib/useSite'

export default defineComponent({
  props: {
    topic: {
      type: [String],
      required: true
    },
    link: {
      type: [String],
      required: false
    },
    sub: {
      type: [String],
      required: false
    },
    sublink: {
      type: [String],
      required: false
    }
  },
  setup (props) {
    const { site } = useSite()
    const titleColorClass = computed(() => site.value.titleColorClass)
    return { titleColorClass }
  }
})
</script>

<style lang="scss" scoped>
div.tabtitle{
  margin: 0;
  padding: 0;
  h1 {
    font-weight: 300;
    margin: 0;
    padding: 0;
    a {
      text-decoration: none;
    }
  }
  h1+h1 {margin-top: 4px}
  h1.subtopic{
    font-size: 16px;
    line-height: 16px;
  }
  h1.topic{
    font-size: 32px;
    line-height: 32px;
  }
}
</style>
