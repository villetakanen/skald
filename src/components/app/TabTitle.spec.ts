import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import TabTitle from './TabTitle.vue'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('Mounted App', () => {
  const wrapper = shallowMount(TabTitle, {
    propsData: {
      topic: 'skald'
    },
    localVue,
    router
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
