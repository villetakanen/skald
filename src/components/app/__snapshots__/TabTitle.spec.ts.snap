// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App renders correctly 1`] = `
<div
  class="tabtitle"
>
  <!---->
   
  <h1
    class="topic "
  >
    skald 
    <!---->
  </h1>
</div>
`;
